esy-build-package/Run.re

```diff
@@ -225,7 +225,15 @@ let copyContents = (~from, ~ignore=[], dest) => {
         Path.Set.empty,
         ignore,
       );
-    `Sat(path => Ok(!Path.Set.mem(path, ignoreSet)));
+    `Sat(
+      path => {
+        let stats = Unix.lstat(Fpath.to_string(path));
+        switch (stats.st_kind) {
+        | Unix.S_LNK => Ok(false)
```

Comment:
I dont think we can simply ignore symlinks. We'll have to recreate them at destination. Like `cp -P`,
We're already copying symlinks later in L268. We need to make sure we traverse symlinks too.

@melwyn95 Do you remember why you need to ignore symlinks?

Reply:
@ManasJayanth it was related to this issue, apparently caused by a symlink cycle: https://github.com/mjambon/dune-deps/issues/23#issuecomment-929470493

Reply:
Yes, as @anchpop mentioned this is required to avoid infinite loop in case of symlink cycle.

Also one more interesting thing in this PR, if you see this snippet from master branch https://github.com/esy/esy/blob/master/esy-build-package/Run.re#L250-L269, this snippet is responsible to copy file/directories/symlinks
But since we do a stat `let* stats = Bos.OS.Path.stat(path);`, the case of symlink was never getting executed, because in case of a symlink stat gives the `st_kind` of the content to which the symlink point to , while debugging the dune-deps issue I noticed, there is a [symlink to parent directory](https://github.com/mjambon/dune-deps/blob/master/test/proj/foo/link-to-parent) in dune-deps repo, So while doing the copying esy does not create a symlink it creates/copies a directory (copies the parent directory in this case)


By doing a lstat `let stats = Unix.lstat(Fpath.to_string(path));` I was getting the correct result (symlink getting copied as a symlink)


Why were we not using `lstat` in the first place?

Reply:
> @ManasJayanth it was related to this issue, apparently caused by a symlink cycle: mjambon/dune-deps#23 (comment)

@anchpop @melwyn95 I understand - the whole point of this PR is to avoid infinite loop in case of a cycle :) What I fail to understand, specifically, is why are skipping symlinks altogether in the traversal. I think we could use a `visited` flag to avoid cycles instead, @melwyn95 . What do you think?